reflex/components/sonner/toast.py

```diff
@@ -27,6 +33,50 @@
 toast_ref = Var.create_safe("refs['__toast']")
 
 
+class ToastAction(Base):
+    """A toast action; rendered a button in the toast."""
+
+    label: str
+    on_click: Any
+
+
+@serializer
+def serialize_action(action: ToastAction) -> dict:
+    """Serialize a toast action.
+
+    Args:
+        action: The toast action to serialize.
+
+    Returns:
+        The serialized toast action with on_click formatted to queue the given event.
+    """
+    if action.on_click is not None:
+        events = []
+        on_click_specs = action.on_click
+        if not isinstance(on_click_specs, list):
+            on_click_specs = [on_click_specs]
+        for spec in on_click_specs:
+            if isinstance(spec, EventHandler):
+                specs = [call_event_handler(spec, lambda: [])]
+            elif isinstance(spec, type(lambda: None)):
+                specs = call_event_fn(spec, lambda: [])
+            else:
+                specs = [spec]
+            events.extend(format.format_event(s) for s in specs)
+        on_click = Var.create(
+            f"() => {{queueEvents([{','.join(events)}], socket); processEvent(socket)}}",
```

Comment:
maybe this could be exposed as a formatting method that user that wrap component can use?

Reply:
moved to `rx.utils.format.format_queue_events`

i think this will be really useful for any `rx.call_script`-based API that needs to pass callback functions that trigger reflex events, so good to formalize it.

and since i formalized it, it was also easy to add in `on_dismiss` and `on_auto_close` callbacks too 😎

Reply:
oh nice, i realized i was also using a variant of this code inside `call_script` itself, so i can clean that up too 😄